python/lsst/ts/aos/utils/m1m3/m1m3_aav.py

```diff
@@ -0,0 +1,761 @@
+from lsst.ts.idl.enums.MTM1M3 import DetailedState
+from lsst_efd_client import EfdClient
+from tqdm import tqdm
+
+tqdm.pandas()
+
+
+class AccelerationAndVelocity:
+    """Compute fir of M1M3 acceleration and velocities.
+
+    Parameters
+    ----------
+    config : pathlib.Path
+        Path to configuration files.
+    load_from : pathlib.Path
+        Optional path to load data (from hd5 file). If None, data are
+        collected/computed.
+
+    Attributes
+    ----------
+    azimuths : pd.DataFrame
+        Mount azimuth. Contains actual and demand Position, Velocity and
+        Acceleration.
+    calculated : pd.DataFrame
+        Forces and moments calculated from the fit.
+    coefficients : pd.DataFrame
+        Coefficients for velocity and acceleration transformation.
+    detailed_states : pd.DataFrame
+        Mirror detailed states before and during selected time interval.
+    elevations : pd.DataFrame
+        Mount elevation. Contains actual and demand Position, Velocity and
+        Acceleration.
+    fitter : AccelerationAndVelocityFitter
+        Fitter doing the actual work - fitting data.
+    interpolated : pd.DataFrame
+        Interpolated raw data.
+    intervals : pd.DataFrame
+        Contains start, end and use flag (whenever to use the interval) of
+        intervals suitable for fitting.
+    mirror : pd.DataFrame
+        Mirror forces. Contains interpolated data, with extra X0..11, Y0..99
+        and Z0..155 columns - those contain data derived from hardpoint forces
+        (f[xyz], m[xyz]). Those are the per-mirror (XYZ) forces the algorithm
+        needs to counter-act.
+    raw : pd.DataFrame
+        Raw data. Concatenation of azimuths, elevations and hardpoint forces.
+        Contains null/NaNs, as the three tables are merged together.
+    residuals : pd.DataFrame
+        Residuals of six forces/components.
+    """
+
+    def __init__(
+        self, efd_name: str, config: pathlib.Path, load_from: pathlib.Path | None
+    ):
+        self.efd_name = efd_name
+        self.force_calculator = ForceCalculator(config)
+
+        self.actual_velocity_limit = 0.01
+        self.demand_velocity_limit = 0.01
+        self.detailed_states: pd.DataFrame | None = None
+
+        self.azimuths: pd.DataFrame | None = None
+        self.calculated: pd.DataFrame | None = None
+        self.coefficients: pd.DataFrame | None = None
+        self.detailed_states: pd.DataFrame | None = None
+        self.elevations: pd.DataFrame | None = None
+        self.fitter: AccelerationAndVelocityFitter | None = None
+        self.interpolated: pd.DataFrame | None = None
+        self.intervals: pd.DataFrame | None = None
+        self.mirror: pd.DataFrame | None = None
+        self.raw: pd.DataFrame | None = None
+        self.residuals: pd.DataFrame | None = None
+
+        if load_from is not None:
+
+            def try_load(name: str) -> pd.DataFrame | None:
```

Comment:
is this the right place to put this function? Just wondering

Reply:
Why not? Python allows that, I don't need that function outside of this simple call (yet)